Jobsway2goMvc/Controllers/HomeController.cs

```diff
@@ -7,17 +9,56 @@ namespace Jobsway2goMvc.Controllers
     public class HomeController : Controller
     {
         private readonly ILogger<HomeController> _logger;
+        private readonly ApplicationDbContext _context;
 
-        public HomeController(ILogger<HomeController> logger)
+        public HomeController(ILogger<HomeController> logger, ApplicationDbContext context)
         {
             _logger = logger;
+            _context = context;
         }
 
         public IActionResult Index()
         {
             return View();
         }
 
+        [HttpPost]
+        public async Task<IActionResult> Index(string searchString, SearchModel searchModel)
+        {
+
+            string model = searchModel.ToString();
+
+            var users = from m in _context.Users
+                        select m;
+            var jobs = from m in _context.Jobs
+                       select m;
+            var events = from m in _context.Events
+                         select m;
+            var groups = from m in _context.Groups
+                         select m;
+
```

Comment:
Move each of these lines inside switch cases. We don't have to fetch everything, this is why we're specifying with enum what we're looking for.